File: javaScript/Queue.js


class Node{
	constructor(data){
        this.data = data;
        this.next = null;
    }
}

class Queue{
	constructor(){
        this.head = null;
		this.size = 0;
        this.tail = null;
    }
	push(data){
		let newNode = new Node(data);
		this.size++;
		if(this.head == null){
			this.head = newNode;
            this.tail = newNode;
		}else{
			this.tail.next = newNode;
			this.tail = newNode;
		}
	}
	pop(){
		if (this.head == null){
			this.tail = null;
			return null;
		}
		this.size--;
		let data = this.head.data
		this.head = this.head.next;
		
		return data;
	}
}

module.exports = Queue;

File: Python/Queue.py

class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class Queue:
    def __init__(self):
        self.head = None
        self.tail = None
        self.size = 0

    def enqueue(self, data) -> None:
        new_node = Node(data)

        if self.tail is None:
            self.head = new_node
            self.tail = new_node
            self.size += 1
            return

        self.size += 1
        self.tail.next = new_node
        self.tail = new_node

    def dequeue(self) -> any:
        if self.size == 0:
            return None

        if self.size == 1:
            data = self.head.value
            self.head = None
            self.tail = None
            self.size = 0
            return data

        data = self.head.value
        self.head = self.head.next
        self.size -= 1
        return data
